src/shared/components/plots/MultipleCategoryBarPlot.tsx

```diff
@@ -913,8 +944,12 @@ export default class MultipleCategoryBarPlot extends React.Component<
         }
     }
 
-    componentDidUpdate() {
+    componentDidUpdate(prevProps: IMultipleCategoryBarPlotProps) {
         this.updateLegendWidth();
+
+        if (this.props.horizontalBars !== prevProps.horizontalBars) {
```

Comment:
@gblaih should be better to use the horizontalBars value in the component "key" attribute.  This will force it to re-mount when that value changes and so you can avoid dirtiness like the above.  You might lose other state, but it's a cleaner approach.  Give it a try.

Reply:
and apologies if this was not your change